### PR TITLE
Refine calling handlers

### LINKED ISSUE #145: Error after value repeats output
``` r
evaluate::evaluate("1;stop('x')")
#> [[1]]
#> $src
#> [1] "1;stop('x')"
#> 
#> attr(,"class")
#> [1] "source"
#> 
#> [[2]]
#> [1] "[1] 1\n"
#> 
#> [[3]]
#> <simpleError in eval(expr, envir, enclos): x>
#> 
#> [[4]]
#> [1] "[1] 1\n"
```

<sup>Created on 2024-06-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by hadley**
Fixes #145

### CODE REVIEW

**Comment by hadley** [comment]
@lionel let me know if you see a simple way to wrap `handle` into `handlers`, otherwise I'll tackle that in a future PR.

File: R/eval.R [hunk 1]
```diff
@@ -248,18 +248,15 @@ evaluate_call <- function(call,
   }
 
   user_handlers <- output_handler$calling_handlers
+  evaluate_handlers <- list(error = eHandler, warning = wHandler, message = mHandler)
+  handlers <- c(user_handlers, evaluate_handlers)
```
Comment (lionel-) [inline R/eval.R:228]:
```suggestion

  # The user's condition handlers have priority over ours
  handlers <- c(user_handlers, evaluate_handlers)
```

**Review by lionel- — APPROVED** [review]

File: R/eval.R [hunk 2]
```diff
@@ -208,10 +208,12 @@ evaluate_top_level_expression <- function(exprs,
...
   }
 
-  ev <- list(value = NULL, visible = FALSE)
-
   if (use_try) {
-    handle <- function(f) try(f, silent = TRUE)
+    handle <- function(code) {
+      tryCatch(code, error = function(err) {
+        list(value = NULL, visible = FALSE)
+      })
```
Comment (lionel-) [inline R/eval.R:215]: It does seem like it would be easier to understand the return type if this was closer to the withVisible call

Comment (hadley) [reply R/eval.R:215]: I'll simplify this further after #153 since it turns out that we don't actually need to capture the visibility of the printed value.

**Pull request merged**